llama.cpp

```diff
@@ -3019,28 +3101,37 @@ struct llama_model_loader {
     }
 
     const char * get_tensor_name(int i) const {
-        return gguf_get_tensor_name(ctx_gguf, i);
+        return weights.at(i).tensor->name;
+    }
+
+    const llama_tensor_weights & get_weights(const char * name) const {
+        for (const auto & weight : weights) {
+            if (strcmp(name, weight.tensor->name) == 0) {
+                return weight;
+            }
+        }
+        throw std::runtime_error(format("tensor %s not found", name));
     }
 
     struct ggml_tensor * get_tensor_meta(const char * name) const {
-        return ggml_get_tensor(ctx_meta, name);
+        return get_weights(name).tensor;
```

Comment:
Handle optional tensors:

```suggestion
        try {
            return get_weights(name).tensor;
        } catch (const std::runtime_error & e) {
            return NULL;
        }
```